Source/InvocationCollection.cs

```diff
@@ -78,46 +78,101 @@ public int Count
 
 		public void Add(Invocation invocation)
 		{
-			lock (this.invocations)
+			lock (this.invocationsLock)
 			{
-				this.invocations.Add(invocation);
+				if (this.count == this.capacity)
+				{
+					var targetCapacity = this.capacity == 0 ? 4 : (this.capacity * 2);
+					Array.Resize(ref this.invocations, targetCapacity);
+					this.capacity = targetCapacity;
+				}
+
+				this.invocations[this.count] = invocation;
+				this.count++;
 			}
 		}
 
 		public bool Any()
 		{
-			return this.invocations.Count > 0;
+			lock (this.invocationsLock)
+			{
+				return this.count > 0;
+			}
```

Comment:
This lock isn't necessary (for the same reason mentioned at `Count`), please remove.

(In fact, now that we're implementing `IEnumerable`, I wonder if this method is still needed at all: We're only using it in a single place&mdash;`VerifyNoOtherCalls`&mdash;which in typical user code is perhaps run on the order of once per unit test method, so we could probably affort just letting Moq use LINQ's `.Any()` operator and going through `.GetEnumerator()`. What do you think?)

Reply:
I think LINQ's `Any()` might just call `Count` in this case. It's probably fine to remove the method.

Reply:
That's fine with me, let's remove it.